src/SearchView.vala

```diff
@@ -179,5 +198,54 @@ public class Switchboard.SearchView : Gtk.Box {
 
             child = grid;
         }
+
+        private static string markup_string_with_search (string text, string pattern) {
+            const string MARKUP = "%s";
+
+            if (pattern == "") {
+                return MARKUP.printf (Markup.escape_text (text));
+            }
+
+            // if no text found, use pattern
+            if (text == "") {
+                return MARKUP.printf (Markup.escape_text (pattern));
+            }
+
+            var matchers = Synapse.Query.get_matchers_for_query (
```

Comment:
Seems a bit overkill maybe :sweat_smile:

Reply:
I copied the code from the application menu 😅, but I will look for a way to optimize the code.